es-ds-components/components/es-verification-code.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import InputOtp from 'primevue/inputotp';
+
+const model = defineModel<string>();
+const verificationCode = ref();
+
+const props = defineProps({
+    charCount: {
+        type: Number,
+        default: 5,
+    },
+});
+
+const count = ref(props.charCount);
+watch(model, (newVal) => {
+    verificationCode.value = newVal;
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const updateCode = (newValue: any) => {
+    model.value = newValue;
+};
+</script>
+
+<template>
+    <div>
+        <input-otp
+            v-model="verificationCode"
+            class="custom-otp"
+            :length="count"
+            integer-only
+            @update:model-value="updateCode">
+            <template #default="{ attrs, events, index }">
+                <input
+                    :id="index.toString()"
+                    type="text"
+                    class="custom-otp-input"
+                    v-bind="attrs"
+                    v-on="events" />
+            </template>
+        </input-otp>
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@use '@energysage/es-ds-styles/scss/variables' as variables;
+@use '@energysage/es-ds-styles/scss/mixins/breakpoints' as breakpoints;
+
+.custom-otp {
+    display: flex;
+    gap: 10px;
+}
+.custom-otp-input {
```

Comment:
If we're moving this CSS here from `es-ds-styles/scss/vue/components/_verification-code.scss` so that it doesn't get unnecessarily included in an app unless the app uses this component, we should probably also delete that `_verification-code.scss` file so we can actually realize those file size savings.